src/SML.py

```diff
@@ -1074,7 +1074,6 @@ def __init__(self, type: str, x: np.ndarray = None,
     self.x = DataSet(name="x", nrows=1, ncols=1)
     if xfile is not None and x is None:
       self.xt.points = np.loadtxt(xfile, skiprows=1, delimiter=",")
-      self.xt.points = self.x.points.reshape(self.x.points.shape[0], -1)
```

Comment:
I don't understand the point of this line
`self.x.points` is empty on initialization so this will always return an error

Reply:
This is a bug, and it lived here since I added a new feature for injecting new data to the training set. However, reshaping extracted data is still required so line 1076 and 1077 should be replaced with the following two lines.

temp = copy.deepcopy(np.loadtxt(xfile, skiprows=1, delimiter=","))
self.xt.points = temp.reshape(temp.shape[0], -1)